crates/cext/Cargo.toml

```diff
@@ -17,10 +17,12 @@ workspace = true
 thiserror.workspace = true
 num-complex.workspace = true
 qiskit-accelerate.workspace = true
+pyo3.workspace = true
```

Comment:
This needs to move to the feature so we only directly depend on pyo3 in pyyhon_binding mode

Reply:
But implicitly we'd still depend on pyo3 through qiskit-accelerate, until we move the Rust-only code into a separate crate, right?

Reply:
`cext` itself only depends on `pyo3` if the Python feature is active. It happens to have transitive dependencies that depend on `pyo3` anyway, but that's an implementation detail as far as it's concerned.